test/test_auth.py

```diff
@@ -74,8 +77,16 @@ def __init__(self, collection):
         self.success = False
 
     def run(self):
-        assert self.collection.find_one({"$where": delay(1)}) is not None
-        self.success = True
+        if _IS_SYNC:
```

Comment:
Since the async client doesn't support multithreading, we don't need to have `AutoAuthenticateThread` support async.